Spaghettification.ipynb

```diff
@@ -0,0 +1,2375 @@
+   },
+   "source": [
+    "Submit your written work via **Gradescope**."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ojiZf6Hccv40"
+   },
+   "source": [
+    "## 2. Nonlinear equation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "R4ufgGbOfNaG"
+   },
+   "source": [
+    "### 2.1. Define a nonlinear equation\n",
+    "\n",
+    "Combining Newton’s law of universal gravitation and Young's modulus, write a single, nonlinear equation in canonical form, $c(r_1,Δ𝐿) = 0$. \n",
+    "\n",
+    "*Hint*: Just like a tidal force, the tensile force on the bus will be generated by the difference in gravitational force between the front of the bus (nearer to the black hole) and the back of the bus (further from the black hole)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "hHm34P-tfvO3"
+   },
+   "source": [
+    "Submit your written work via **Gradescope**."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### BEGIN SOLUTION\n",
+    "\"\"\"\n",
+    "0 = ((L_0 * G * m_1 * m_2) / (Y * A)) * ((1 / r^2) - (1 / (r + L_0 + ΔL)^2)) - ΔL\n",
+    "\"\"\"\n",
+    "### END SOLUTION"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "CqEWYdt0f76r"
+   },
+   "source": [
+    "### 2.2. Define a function that returns the residuals\n",
+    "\n",
+    "Using the equation you derived in **2.1**, write a Python function that returns its residual given some value of $r_1$ and some value of $\\Delta L$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 130
+    },
+    "id": "Iq6ayqYhD9jZ",
+    "outputId": "bd3cf883-a0ef-414c-c852-c852276836e2"
+   },
+   "outputs": [],
+   "source": [
+    "# function for nonlinear equation residual\n",
+    "def residual_function(r_1,ΔL):\n",
+    "    ''' Returns the residual of the nonlinear equation given in canonical form\n",
+    "    \n",
+    "    Args:\n",
+    "        r_1, ΔL\n",
+    "        \n",
+    "    Returns:\n",
+    "        residual    \n",
+    "    '''\n",
+    "  # Add your solution here\n",
+    "    residual = \n",
+    "    return residual"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {
+    "id": "bDXosTblE8zc"
+   },
+   "outputs": [],
+   "source": [
+    "### BEGIN SOLUTION\n",
+    "# function for nonlinear equation residual\n",
+    "def residual_function(r_1,ΔL):\n",
+    "    ''' Returns the residual of the nonlinear equation given in canonical form\n",
+    "  \n",
+    "  Args:\n",
+    "      r_1, ΔL\n",
+    "      \n",
+    "  Returns:\n",
+    "      residual    \n",
+    "  '''\n",
+    "  # Add your solution here\n",
+    "    residual =  ((L_0 * G * m_1 * m_2) / (Y * A)) * ((1 / r_1**2) - (1 / (r_1 + L_0 + ΔL)**2)) - ΔL\n",
```

Comment:
Move the ### BEGIN SOLUTION ### and ### END SOLUTION ### to above and below this line.